src/adapter/src/optimize/dataflows.rs

```diff
@@ -576,18 +578,17 @@ fn eval_unmaterializable_func(
     };
 
     match f {
-        UnmaterializableFunc::CurrentDatabase => pack(Datum::from(session.vars().database())),
+        UnmaterializableFunc::CurrentDatabase => pack(Datum::from(session.database())),
         UnmaterializableFunc::CurrentSchema => {
-            let catalog = state.for_session(session);
-            let schema = catalog
-                .search_path()
+            let search_path = state.resolve_search_path(session);
+            let schema = search_path
                 .first()
                 .map(|(db, schema)| &*state.get_schema(db, schema, session.conn_id()).name.schema);
             pack(Datum::from(schema))
         }
```

Comment:
According to the coverage information, this is not covered by a test case. (Though, I am aware that you only conducted a minor change here.)
If it is easy, could you add a test case?